packages/nuxt/src/app/composables/state.ts

```diff
@@ -35,3 +36,23 @@ export function useState <T> (...args: any): Ref<T> {
   }
   return state
 }
+
+export function clearNuxtState (
+  keys?: string | string[] | ((key: string) => boolean)
+): void {
+  const nuxtApp = useNuxtApp()
+  const _allKeys = Object.keys(nuxtApp.payload.state)
+  const _keys: string[] = !keys
+    ? _allKeys
+    : typeof keys === 'function'
+      ? _allKeys.filter(keys)
+      : Array.isArray(keys)
+        ? keys.map(k => useStateKeyPrefix + k)
+        : [useStateKeyPrefix + keys]
+
+  for (const key of _keys) {
+    if (key in nuxtApp.payload.state) {
+      nuxtApp.payload.state[key] = undefined
+    }
+  }
```

Comment:
slightly more performant not to do another full iteration

```suggestion
  const _keys: string[] = !keys
    ? _allKeys
    : typeof keys === 'function'
      ? _allKeys.filter(keys)
      : Array.isArray(keys) ? keys : [keys]

  for (const key of _keys) {
    if (key in nuxtApp.payload.state) {
      nuxtApp.payload.state[useStateKeyPrefix + key] = undefined
    }
  }
```